core/c4diagram.go

```diff
@@ -0,0 +1,355 @@
+// package defines diagram as code using the graph input
+package core
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/kislerdm/diagramastext/core/compression"
+)
+
+type optionsPlantUMLClient struct {
+	httpClient HttpClient
+}
+
+type clientPlantUML struct {
+	options optionsPlantUMLClient
+
+	baseURL string
+}
+
+func (c *clientPlantUML) Do(v DiagramGraph) (ResponseDiagram, error) {
+	code, err := diagramGraph2plantUMLCode(v)
+	if err != nil {
+		return nil, err
+	}
+	p, err := code2Path(code)
+	if err != nil {
+		return nil, err
+	}
+
+	diagram, err := c.requestHandler("svg/" + p)
+	if err != nil {
+		return nil, err
+	}
+
+	return ResponseC4Diagram{SVG: *(*string)(unsafe.Pointer(&diagram))}, nil
+}
+
+func (c *clientPlantUML) requestHandler(p string) ([]byte, error) {
```

Comment:
**suggestion** I miss a lot here some debug/log options, what do you think about using the `httputil.DumpRequest` and `httputil.DumpResponse`?

Reply:
It may be useful indeed, yet I'd appreciate if you could share your rational. `I miss a lot here` is quite subjective, don't you think? :) Please make a suggestion using the github review functionality.